### PR TITLE
sys: cleanup posix compat code

### PR DESCRIPTION
**Pull request by kaspar030**
- removed posix_io module. Since the removal of chardev_thread, it was probably completely unnused.
- the posix module had it's own "unistd.h" header, which clashed with cpu-scope or newlib files. removed it.
(the file contained macro hackery for OS X. I tested this PR on milhouse, works fine.) *edit* moved it to cpu libc header override directories *edit*
- implemented sleep/usleep using xtimer (previously: vtimer)
- fixed tests to no disable auto_init (as that would leave xtimer uninitialized)

### CODE REVIEW

**Comment by kaspar030** [comment]
(enabling travis to see build errors)

**Comment by cgundogan** [comment]
nice +87 −312

**Comment by kaspar030** [comment]
CI seems fine, so I squashed the hotfixes. Sorry for not marking WIP before, but now it's ready for reviewing.

I'm aware that unistd.h is duplicate for both msp430 and avr, but I couldn't come up with a simple way of sharing that single file.

**Comment by kaspar030** [comment]
Travis is fine...

File: cpu/atmega_common/avr-libc-extra/unistd.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#ifndef UNISTD_H_
+#define UNISTD_H_
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define STDOUT_FILENO (0)
```
Comment (Kijewski) [inline cpu/atmega_common/avr-libc-extra/unistd.h]: Probably not really needed, but the old (correct) definition won't hurt, will it?
```c
#define STDIN_FILENO    0   ///< stdin file descriptor
#define STDOUT_FILENO   1   ///< stdout file descriptor
#define STDERR_FILENO   2   ///< stderr file descriptor
```

Comment (kaspar030) [inline cpu/atmega_common/avr-libc-extra/unistd.h]: yep. fixed and squashed immediately.

**Comment by kaspar030** [comment]
Hm, when I click on the supposedly failed travis result, everything is green. @Kijewski would you ACK?

**Comment by Kijewski** [comment]
ACK, Travis is happy and the diff looks sane.

**Comment by kaspar030** [comment]
thx for reviewing!

**Pull request merged**